### PR TITLE
Ignore body-prose rule when paragraph ends with url

### PR DESCRIPTION
**Pull request by realmarv**
Fixes https://github.com/nblockchain/conventions/issues/45

### CODE REVIEW

File: commitlint.config.ts [hunk 1]
```diff
@@ -219,6 +221,7 @@ module.exports = {
                                 }
                                 return false;
                             }
+                            console.log("=bodyStr=>" +  bodyStr + "<==")
```
Comment (knocte) [inline commitlint.config.ts:224]: @realmarv ^

File: commitlint.config.ts [hunk 2]
```diff
@@ -231,6 +234,7 @@ module.exports = {
                                 let startWithLowerCase = isLowerCase(paragraph[0]);
 
                                 let validParagraphEnd = paragraphHasValidEnding(paragraph);
+                                console.log("=Paragraph=>" + validParagraphEnd + " " + paragraph + "<==")
```
Comment (knocte) [inline commitlint.config.ts:237]: @realmarv ^

File: commitlint.config.ts [hunk 3]
```diff
@@ -206,11 +206,13 @@ module.exports = {
...
 
                             function paragraphHasValidEnding(paragraph: string): boolean {
+                                let lastWordInParagraph = paragraph.split(' ')[paragraph.split(' ').length - 1]
                                 let endingChar = paragraph[paragraph.length - 1];
                                 if (endingChar === '.' ||
                                     endingChar === ':' ||
                                     endingChar === '!' ||
-                                    endingChar === '?') {
+                                    endingChar === '?' ||
+                                    isValidUrl(lastWordInParagraph)) {
```
Comment (knocte) [inline commitlint.config.ts:215]: it is better and more readable to call isValidUrl() before the `if` that checks endingChar

Comment (knocte) [reply commitlint.config.ts:215]: @realmarv

File: commitlint.config.ts [hunk 4]
```diff
@@ -206,11 +206,13 @@ module.exports = {
                         if (bodyStr !== ''){
 
                             function paragraphHasValidEnding(paragraph: string): boolean {
+                                let lastWordInParagraph = paragraph.split(' ')[paragraph.split(' ').length - 1]
```
Comment (knocte) [inline commitlint.config.ts:209]: please end typescript lines with semicolon; and don't call paragraph.split(' ') twice

File: commitlintplugins.test.ts [hunk 5]
```diff
@@ -160,6 +160,19 @@ test('body-prose15', () => {
 });
 
 
+test('body-prose16', () => {
+    let commitMsgWithUrlAtTheEndOfParagraph =
+        "input: Frontend.XF.Android: switch to SDK-style\n\n" +
```
[the remark this answers is not in the document]
Comment (knocte) [reply commitlintplugins.test.ts:165]: (That was just a bad copy-paste from my side, the error message from commitlint includes the `input:` thing, that was not part of the commit message)

[the remark this answers is not in the document]
Comment (realmarv) [reply commitlintplugins.test.ts]: Done

File: commitlint.config.ts (hunk 4 above)
Comment (realmarv) [reply commitlint.config.ts]: Done

File: commitlint.config.ts (hunk 3 above)
Comment (knocte) [reply commitlint.config.ts]: @realmarv sorry this is not done, you're still calling isValidUrl() after checking endingChar; you should be able to call it before (even before you assign endingChar at all)

Comment (realmarv) [reply commitlint.config.ts]: You mean like this?

Comment (knocte) [reply commitlint.config.ts]: yes, that looks way more readable and maintainable, thanks

Comment (realmarv) [reply commitlint.config.ts]: Great!

**Comment by knocte** [comment]
I'm going to merge this, but next time, try to be more verbose in commit msgs please. I say this because "fix the rule" is not enough. If someone finds your commit in the git history, "fix the rule" doesn't give any clues about what rule and what was broken (I'm still merging it, hoping that finding the link between this commit and the Merge commit is easy; but from now on, let's improve our commit messages please).

**Comment by realmarv** [comment]
> I'm going to merge this, but next time, try to be more verbose in commit msgs please. I say this because "fix the rule" is not enough. If someone finds your commit in the git history, "fix the rule" doesn't give any clues about what rule and what was broken (I'm still merging it, hoping that finding the link between this commit and the Merge commit is easy; but from now on, let's improve our commit messages please).

Right. I'll keep that in mind.

**Pull request merged**